docs/gallery/domain/plot_icephys.py

```diff
@@ -0,0 +1,703 @@
+#           data from the :py:class:`~pynwb.base.TimeSeriesReferenceVectorData`, the missing values
+#           will be represented via masked numpy arrays, i.e., as masked values in a
+#           ``numpy.ma.masked_array`` or as a ``np.ma.core.MaskedConstant``.
+
+rowindex3 = nwbfile.add_intracellular_recording(
+    electrode=electrode,
+    response=response,
+    id=12
+)
+
+#####################################################################
+# .. warning:: For brevity we reused in the above example the same response and stimulus in
+#        all rows of the intracellular_recordings. While this is allowed, in most practical
+#        cases the stimulus and response will change between intracellular_recordings.
+
+#####################################################################
+# Adding custom columns to the intracellular recordings table
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# We can add a column to the main intracellular recordings table as follows.
+
+nwbfile.intracellular_recordings.add_column(
+    name='recording_tag',
+    data=['A1', 'A2', 'A3'],
+    description='String with a recording tag'
+)
+
+#####################################################################
+# The :py:class:`~pynwb.icephys.IntracellularRecordingsTable` table is not just a ``DynamicTable``
+# but an ``AlignedDynamicTable`. The ``AlignedDynamicTable` type is itself a ``DynamicTable``
+# that may contain an arbitrary number of additional ``DynamicTable``, each of which defines
+# a "category". This is similar to a table with "sub-headings". In the case of the
+# :py:class:`~pynwb.icephys.IntracellularRecordingsTable`, we have three predefined categories,
+# i.e., electrodes, stimuli, and responses. We can also dynamically add new categories to
+# the table. As each category corresponds to a DynamicTable, this means we have to create a
+# new DynamicTable and add it to our table.
+
+# Create a new DynamicTable for our category that contains a location column of type VectorData 
+location_column = VectorData(
+    name='location',
+    data=['Mordor', 'Gondor', 'Rohan'],
+    description='Recording location in Middle Earth'
+)
+
+lab_category = DynamicTable(
+    name='recording_lab_data',
+    description='category table for lab-specific recording metadata',
+    colnames=['location', ],
+    columns=[location_column, ]
+)
+# Add the table as a new category to our intracellular_recordings
+nwbfile.intracellular_recordings.add_category(category=lab_category)
+# Note, the name of the category is name of the table, i.e., 'recording_lab_data'
+
+#####################################################################
+# .. note:: In an ``AlignedDynamicTable`` all category tables MUST align with the master table,
+#           i.e., all tables must have the same number of rows and rows are expected to
+#           correspond to each other by index
+
+#####################################################################
```

Comment:
This section was previously before the above section, but I think it makes more sense after the above section because it references subcategory tables, which are introduced only in the above section.

Reply:
Sounds good